main/src/library/Reducible.flix

```diff
@@ -0,0 +1,183 @@
+
+    ///
+    /// Returns the last element of `t`.
+    ///
+    pub def last(t: t[a]): a =
+        Reducible.reduceRight((_, acc) -> acc, t) // Could use reduceLeft w. lambda (_, a) -> a
+
+    ///
+    /// Returns the number of elements in `t` that satisfy the predicate `f`.
+    ///
+    pub def count(f: a -> Bool, t: t[a]): Int32 =
+        Reducible.reduceMap(a -> if (f(a)) 1 else 0, t)
+
+    ///
+    /// Returns the number of elements in `t`.
+    ///
+    pub def length(t: t[a]): Int32 =
+        Reducible.reduceMap(_ -> 1, t)
+
+    ///
+    /// Returns the sum of all elements in `t`.
+    ///
+    pub def sum(t: t[Int32]): Int32 = ???
+
+    ///
+    /// Returns the sum of all elements in `t` according to the function `f`.
+    ///
+    pub def sumWith(f: a -> Int32 & ef, t: t[a]): Int32 = ???
+
+    ///
+    /// Returns the product of all elements in `t`.
+    ///
+    pub def product(t: t[Int32]): Int32 = ???
+
+    ///
+    /// Returns the product of all elements in `t` according to the function `f`.
+    ///
+    pub def productWith(f: a -> Int32 & ef, t: t[a]): Int32 = ???
+
+    ///
+    /// Returns `true` if and only if at least one element in `t` satisfies the predicate `f`.
+    ///
+    pub def exists(f: a -> Bool & ef, t: t[a]): Bool = ???
+
+    ///
+    /// Returns `true` if and only if all elements in `t` satisfy the predicate `f`.
+    ///
+    pub def forall(f: a -> Bool & ef, t: t[a]): Bool = ???
+
+    ///
+    /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from left to right.
+    ///
+    pub def findLeft(f: a -> Bool & ef, t: t[a]): Option[a] = ???
+
+    ///
+    /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from right to left.
+    ///
+    pub def findRight(f: a -> Bool & ef, t: t[a]): Option[a] = ???
+
+    ///
+    /// Returns `true` if and only if the element `x` is in `t`.
+    ///
+    pub def memberOf(x: a, t: t[a]): Bool with Eq[a] = ???
+
+    ///
+    /// Finds the smallest element of `t` according to the `Order` on `a`.
+    ///
+    pub def minimum(t: t[a]): a with Order[a] = ???
+
+    ///
+    /// Finds the smallest element of `t` according to the given comparator `cmp`.
+    ///
+    pub def minimumBy(cmp: (a, a) -> Comparison & ef, t: t[a]): a = ???
+
+    ///
+    /// Finds the largest element of `t` according to the `Order` on `a`.
+    ///
+    pub def maximum(t: t[a]): a with Order[a] = ???
+
+    ///
+    /// Finds the largest element of `t` according to the given comparator `cmp`.
+    ///
+    pub def maximumBy(cmp: (a, a) -> Comparison & ef, t: t[a]): a = ???
+
+    ///
+    /// Returns `t` as a non-empty list.
+    ///
+    pub def toNel(t: t[a]): Nel[a] = ???
+
+    ///
+    /// Returns `t` as a list.
+    ///
+    pub def toList(t: t[a]): List[a] = ???
+
+    ///
+    /// Returns `t` as a set.
+    ///
+    pub def toSet(t: t[a]): Set[a] with Order[a] = ???
+
+}
```

Comment:
+ foreach

See also what is in foldable, e.g.:

+ init (returns list)
+ tail (returns list)
+ dropWhile
+ takeWhile
+ toXYZs
+ intersperse?

Maybe come up with a list from Foldable and from Nel and we can discuss.

Reply:
Sure sounds good